### PR TITLE
Bug1277420 - show friendly prompt when cancelling a completed build

### PR DESCRIPTION
**Pull request by jhadvig**
Based on https://bugzilla.redhat.com/show_bug.cgi?id=1277420

I've remove the verbosity level from all the spots in `cancelbuild.go` since the output of all of them is important for the user.

@fabianofranz PTAL

### CODE REVIEW

**Comment by fabianofranz** [comment]
@jhadvig We use `fmt.Fprint*` to display messages to users, and `glog` only for logging with verbosity. So you should use `fmt` here, with the `io.Writer` the command receives. [Example](https://github.com/jhadvig/origin/blob/BZ_1277420/pkg/cmd/cli/cmd/startbuild.go#L378). ;)

**Comment by jhadvig** [comment]
@fabianofranz yep at first I wanted to do it with the `fmt.Fprintf` but I've found out that we also use the `isBuildCancellable` method in the https://github.com/openshift/origin/blob/master/pkg/build/controller/controller.go#L46 without `io.Writer`, so I wasn't sure if I should create a new one, but I guess it's ok

**Comment by jhadvig** [comment]
sorry now I see that the `isBuildCancellable` is declared for the controller.go separatelly. My bad

**Comment by jhadvig** [comment]
@fabianofranz updated, ptal

**Comment by fabianofranz** [comment]
Probably worth adding a couple tests. `test/cmd/builds.sh`, which is called by `hack/test-cmd.sh`, have a few calls to `cancel-build` already. Adding some `grep` to the existing calls to make sure we output the correct messages should suffice.

**Comment by jhadvig** [comment]
@fabianofranz thanks for suggestion. Updated and rebased :)

**Comment by fabianofranz** [comment]
LGTM [test]

**Comment by jhadvig** [comment]
@fabianofranz the test are passing. Unfortunately I had to use the the variant you have been suggesting, since mine was failing.

**Comment by fabianofranz** [comment]
[merge]

**Comment by bparees** [comment]
@stevekuznetsov so it did use to work.......

**Comment by stevekuznetsov** [comment]
@bparees Yes, somehow it did.

File: test/cmd/builds.sh [hunk 1]
```diff
@@ -58,11 +58,21 @@ frombuild=$(oc start-build --from-build="${started}")
...
+os::cmd::expect_success "oc cancel-build ${started} --dump-logs --restart | grep 'Restarted build ${started}.'"
 os::cmd::expect_success 'oc delete all --all'
 os::cmd::expect_success 'oc process -f examples/sample-app/application-template-dockerbuild.json -l build=docker | oc create -f -'
 os::cmd::expect_success "tryuntil 'oc get build/ruby-sample-build-1'"
-# Uses type/name resource syntax
-os::cmd::expect_success 'oc cancel-build build/ruby-sample-build-1'
+# Uses type/name resource syntax to cancel the build and check for proper message
+os::cmd::expect_success "oc cancel-build build/ruby-sample-build-1 | grep 'Build build/ruby-sample-build-1 was cancelled.'"
+# Make sure canceling already cancelled build returns proper message
+os::cmd::expect_success "tryuntil $(oc cancel-build build/ruby-sample-build-1 | grep 'A cancellation event was already triggered for the build build/ruby-sample-build-1.')"
```
Comment (stevekuznetsov) [inline test/cmd/builds.sh]: @jhadvig FYI this should be `os::cmd::try_until_success`

Comment (stevekuznetsov) [inline test/cmd/builds.sh]: @jhadvig How does this test pass for you locally? The output from the API server does not name the build `build/<buildname>`, so this `grep` does not work for me locally.

**Pull request merged**